msgraph/command_modules/profile/custom.py

```
@@ -3,42 +3,77 @@
 # Copyright (c) Microsoft Corporation. All rights reserved.
 # Licensed under the MIT License. See License.txt in the project root for license information.
 # --------------------------------------------------------------------------------------------
-import json
+'''
+Update the _help.py file after changing the signature or behavior of functions in this file
+'''
 from knack.prompting import prompt_choice_list
 from msgraph.cli.core.constants import DEFAULT_CLOUDS
-from msgraph.cli import read_profile, write_profile
+from ._cloud_manager import CloudManager
+
+cloud_manager = CloudManager()
+
+
+def current_cloud():
+    return cloud_manager.get_current_cloud() or DEFAULT_CLOUDS['PUBLIC']
+
+
+def delete_cloud(name: str):
+    cloud_manager.delete_cloud(name)
+    print(f'Cloud "{name}" deleted successfully')
+
+
+def add_cloud(name: str, graph_endpoint: str, azure_ad_endpoint: str):
+    _validate(graph_endpoint)
+
+    properties = {
+        'name': name,
+        'graph_endpoint': graph_endpoint,
+        'azure_ad_endpoint': azure_ad_endpoint
+    }
+
+    cloud_manager.create_cloud(name, properties)
+    print(f'Cloud "{name}" added successfully')
```

`create_cloud()` should return status about whether the cloud was added successfully. Use the status as input to `print`.

---

@MIchaelMainer and I had a conversation about this and we decided to throw an error instead of returning a status. This will involve parameterizing write_profile so that we can pass custom error messages to it.